crates/pera-core/src/authority.rs

```diff
@@ -1522,6 +1531,41 @@ impl AuthorityState {
         Ok(())
     }
 
+    /// Handle the MPC signature events emitted from the transaction, if any.
+    /// The filtering to handle only signature-mpc related events happens within [`SignatureMPCManager::handle_mpc_events`] function.
+    async fn handle_signature_mpc_events(
+        &self,
+        inner_temporary_store: &InnerTemporaryStore,
+        effects: &TransactionEffects,
+        epoch_store: &Arc<AuthorityPerEpochStore>,
+    ) -> anyhow::Result<()> {
+        if !self.is_validator(epoch_store) {
+            return Ok(());
+        }
+        let status = match &effects {
+            TransactionEffects::V1(effects) => effects.status(),
+            TransactionEffects::V2(effects) => effects.status(),
+        };
+        if status.is_ok() {
+            let mut signature_mpc_manager = epoch_store.proof_mpc_manager.get();
+            match signature_mpc_manager {
+                Some(mpc_manager) => {
+                    let mut mpc_manager = mpc_manager.lock().await;
+                    mpc_manager.handle_mpc_events(&inner_temporary_store.events.data)
+                }
+                None => {
+                    // This function is being executed for all events, some events are being emitted before the MPC manager is initialized.
+                    // TODO (#250): Make sure that the MPC manager is initialized before any MPC events are emitted.
+                    info!("MPC manager is not initialized");
+                    Ok(())
+                }
+            }
```

Comment:
Aren't there more idomatic ways to match on the variants of `Option` in Rust?

Reply:
There are, it will be refactored.